cache/http.go

```diff
@@ -90,22 +105,36 @@ func ensureDirExists(path string) {
 func (h *httpCache) CacheHandler(w http.ResponseWriter, r *http.Request) {
 	defer r.Body.Close()
 
+	// Helper function for logging responses
+	logResponse := func(code int) {
+		// Parse the client ip
+		clientIp, _, err := net.SplitHostPort(r.RemoteAddr)
+		if err != nil {
+			h.errorLogger.Fatal("Failed parsing RemoteAddr %s: %s", r.RemoteAddr, err.Error())
```

Comment:
Use `h.errorLogger.Fatalf`

Reply:
Actually, I am thinking using `.Fatal` here is wrong. A single request would crash the whole server (this is why go test fails on CI). Can we just log `r.RemoteAddr` as is and thus not even encounter this error case?

Reply:
removed the Fatal call